### PR TITLE
Add: Director Spawning Hordes; + Wave Editor

### PR DESCRIPTION
**Pull request by xxfast**
Implemented horde spawning mechanic and also a custom editor to edit and customize the waves. Its pretty intuitive to work with, so try it out

sowagcmoifusq

### CODE REVIEW

**Comment by xxfast** [comment]
[Director Mechanic - Horde Spawning](https://trello.com/c/wvaN1Fn5/7-director-mechanic-horde-spawning)

File: Assets/Scripts/EnemySpawner.cs [hunk 1]
```diff
@@ -32,7 +48,27 @@ public class EnemySpawner : MonoBehaviour {
...
-			enemy.GetComponent<FollowTarget> ().target = this.gameObject;
+			enemy.GetComponent<FollowTarget> ().target = target;
 		}
 	}
+
+	/// <summary>
+	/// Attachs a profile to the current enemy spawner.
+	/// </summary>
+	/// <param name="toAttach">To attach.</param>
+	public void attachProfile(WaveProfile toAttach){
```
Comment (JPGalovic) [inline Assets/Scripts/EnemySpawner.cs]: Need detail about what each of these do

**Review by JPGalovic — DISMISSED** [review]
System works well and setup is simple enough

Need some details about what each element does.

Spawn Distance, Spawn Freq, Spawn for.

Ideally if possible it would be good to enter details for a wave as both a set of enemy types but a number of each, ie I want to spawn 3 basic and 1 aggressive enemy for the wave.

also is it best to have a time limit between each wave or simply next wave on completion of current wave? not a change but worth a thought.

File: Assets/Scripts/EnemySpawner.cs (hunk 1 above)
Comment (xxfast) [inline Assets/Scripts/EnemySpawner.cs]: done

Comment (xxfast) [reply Assets/Scripts/EnemySpawner.cs]: as of right now, the enemy spawner will spawn enemies assigned in random order.

Comment (xxfast) [reply Assets/Scripts/EnemySpawner.cs]: And about the time between the waves, definitely an interesting approach, but its out of the scope of this task according to the description

**Comment by xxfast** [comment]
I like the ideas you've suggested,  but its out of the scope of this task according to the description. Well work on them as a separate card

**Comment by JPGalovic** [comment]
you still need to indicate what Spawn Distance, Spawn Freq, Spawn For actully do.

In general comment your code as it is not obvious what the above mentioned feilds actully do.

**Review by JPGalovic — CHANGES_REQUESTED** [review]
Comment Spawn Distance, Spawn Freq, Spawn for

needs to be clear what these do.

not beyond scope of task

File: Assets/Scripts/WaveProfile.cs [hunk 2]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections;
```
Comment (JPGalovic) [inline Assets/Scripts/WaveProfile.cs:1]: Cool, so by my understanding the enemy spawns at a frequency of x or y seconds, so to work out number of enemies you take spawnFor and devide by spawnFrequancy

File: Assets/Scripts/WaveProfile.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+	/// <summary>
+	/// Distance from the target to spawn the enemies from
+	/// </summary>
+	public float spawnDistance;
+
+	[Tooltip("Frequency of the enemy spawns in seconds")]
+	/// <summary>
+	/// Frequency of the enemy spawns in seconds
+	/// </summary>
+	public float spanwFrequency;
```
Comment (JPGalovic) [inline Assets/Scripts/WaveProfile.cs:26]: Will need to correct spelling of variable name here but thats for cleanup.

**Review by JPGalovic — APPROVED** [review]
Looks Good.

It is now clear how spawning frequancy works.

Some minor cleanup tasks to complete.

**Pull request merged**